confidential-data-hub/kms/src/plugins/aliyun/credential.rs

```diff
@@ -35,25 +38,30 @@ struct Password {
     client_key_password: String,
 }
 
-impl Credential {
+#[derive(Clone, Debug)]
+pub(crate) struct CredentialEcsRamRole {
+    ecs_ram_role_name: String,
+}
+
+// implement CredentialClientKey related function
+impl CredentialClientKey {
     pub(crate) fn new(client_key: &str, pswd: &str) -> Result<Self> {
         let ck: ClientKey = serde_json::from_str(client_key)?;
 
         let password: Password = serde_json::from_str(pswd)?;
-
         let private_key =
             Self::parse_private_key(ck.private_key_data, password.client_key_password)?;
-
         let private_key = PKey::private_key_from_der(&private_key)?;
-        let credential = Credential {
+
+        let credential = CredentialClientKey {
             client_key_id: ck.key_id.clone(),
             private_key,
         };
 
         Ok(credential)
     }
 
-    pub(crate) fn generate_bear_auth(&self, str_to_sign: &str) -> Result<String> {
+    pub(crate) fn sign(&self, str_to_sign: &str) -> Result<String> {
```

Comment:
This function actually generates bear auth string. Do we need to change the name?

Reply:
I was originally hoping to unify the function names for obtaining signatures in clientKey access and ecsRamRole access. It seems that the `sign()` might be confusing. What do you think about changing it to `get_signature()`?